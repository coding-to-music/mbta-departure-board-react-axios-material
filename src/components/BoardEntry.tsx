import React, { memo } from 'react';
import TableRow from '@material-ui/core/TableRow';
import { withStyles } from '@material-ui/core/styles';
import TableCell from '@material-ui/core/TableCell';
import isEqual from 'react-fast-compare';
import { IPrediction } from '../dataSource/IPredition';

import {
  getCarrier,
  getTrainNumber,
  getTrainStatus,
  getTrainTime,
  getTrainTrackNumber,
} from '../utils/boardData';

export const StyledTableCell = withStyles({
  body: {
    border: 'none',
    textTransform: 'uppercase',
    fontWeight: 'bold',
    fontSize: 20,
    color: '#FFFD5A',
    backgroundColor: '#121312'
  },
})(TableCell);

/**
 * This component represent an entry on the board (Departure or Arrival)
 * @param props Contains a prediction from MTBA API
 * @returns
 */
function BoardEntry(props: { data: IPrediction, isArrival: boolean }) {
  const { data, isArrival } = props;

  const trainNumber = getTrainNumber(data);
  const trainTrackNumber = getTrainTrackNumber(data);
  const status = getTrainStatus(data);
  const trainTime = getTrainTime(data, isArrival);

  return (
    <TableRow className={isArrival ? 'arrival' : 'departure'}>
      <StyledTableCell align="left">{getCarrier()}</StyledTableCell>
      <StyledTableCell align="right">{trainTime}</StyledTableCell>
      <StyledTableCell align="right">{data.destination}</StyledTableCell>
      <StyledTableCell align="right">{trainNumber}</StyledTableCell>
      <StyledTableCell align="right">{trainTrackNumber}</StyledTableCell>
      <StyledTableCell align="right">{status}</StyledTableCell>
    </TableRow>
  );
}

export default memo(
  BoardEntry,
  isEqual
);
