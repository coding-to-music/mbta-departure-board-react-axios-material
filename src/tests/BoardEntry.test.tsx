import React from 'react';
import renderer from 'react-test-renderer';
import BoardEntry from '../components/BoardEntry';
import fakePredictions from './fakePredictions';

test('Board Entry renders correctly', () => {
  const tree = renderer.create(<BoardEntry data={fakePredictions[0]} />).toJSON();
  expect(tree).toMatchSnapshot();
});
