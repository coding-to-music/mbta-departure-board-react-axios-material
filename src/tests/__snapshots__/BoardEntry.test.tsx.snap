// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board Entry renders correctly 1`] = `
<tr
  className="MuiTableRow-root"
  role={null}
>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignLeft"
  >
    MTBA
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignRight"
  >
    1:53 AM
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignRight"
  >
    
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignRight"
  >
    10042
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignRight"
  >
    TBD
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignRight"
  />
</tr>
`;
